dartagnan/src/main/java/com/dat3m/dartagnan/wmm/Wmm.java

```diff
@@ -205,23 +205,28 @@ private Relation makePredefinedRelation(String name) {
             case CTRLDIRECT -> new DirectControlDependency(r);
             case EMPTY -> new Empty(r);
             case IDDTRANS -> new TransitiveClosure(r, getOrCreatePredefinedRelation(IDD));
-            case DATA -> intersection(r,
-                    getOrCreatePredefinedRelation(IDDTRANS),
-                    addDefinition(product(newRelation(), Tag.MEMORY, Tag.MEMORY))
-            );
+            case DATA -> intersection(r, getOrCreatePredefinedRelation(IDDTRANS),
+                    addDefinition(new CartesianProduct(newRelation(),
+                    Filter.union(Filter.byTag(Tag.MEMORY), Filter.byTag(Tag.ALLOC)),
+                    Filter.byTag(Tag.MEMORY))));
```

Comment:
I'm really not a big fan of removing the memory tags from `Alloc` and `Free`. It messes with so many things.
Is there any imaginable case where treating Alloc/Free as proper memory events is really wrong?
If so, then it suggests that `malloc` and `calloc` are ought to be treated differently, since the latter zero-initializes the memory and so almost acts as a write (and hence a proper memory access).

If it is hard to imagine, I would go with the simpler solution and add the tags (or rather, not remove them in the first place) and see if we ever encounter an issue with it.